### ISSUE TITLE
net::ERR_INVALID_ARGUMENT

### ISSUE THREAD

**Issue by theKashey** [cypress-io/cypress#6916]
I would have to skip the issue template, as long as it's not quite useful for my problem.

In short:
- we were using cypress 4.0.2, and everything was ok
- then, without any reasons, we got a few red tests. Every time it was the same problem - the page was loaded only partially. Like networking broken
- we were unable to reproduce it on local machine - only on CI
- After a few ways we just _skipped_ a few tests 😢 
- (today)Several weeks later we got another one

I've tried to update Cypress first to 4.2.0, then to 4.3.0, and in both cases more tests were breaking. With version 4.3.0 I could reproduce the problem locally.

In short - `net::ERR_INVALID_ARGUMENT`.
<img width="411" alt="Screen Shot 2020-04-02 at 5 20 44 pm" src="https://user-images.githubusercontent.com/582410/78217622-2ccc5c00-7508-11ea-8d41-de5bf31e23c8.png">
<img width="975" alt="Screen Shot 2020-04-02 at 5 21 03 pm" src="https://user-images.githubusercontent.com/582410/78217708-4e2d4800-7508-11ea-967c-1dd302f86165.png">

As far as I could see - it could do XHR(before and after), but not able to load scripts dynamically sometimes.
To be a bit more concrete - I could trace the error to the webpacks manifest, when it is prefetches script, but it would also never load that script in the future. If I remove prefetch any script would be just `canceled` on the test end
<img width="586" alt="Screen Shot 2020-04-02 at 5 28 05 pm" src="https://user-images.githubusercontent.com/582410/78217694-47063a00-7508-11ea-881e-eb5250d455fd.png">

### What is strange
We have 3 packs of tests:
- normal tests. Only they had flake during the first encounter
- __the same__ tests in a bit different form. They never had any flakes during the first encounter, and got completely broken (>50%) after bumping Cypress version.
- other tests, for the same application, which never had any issues.

## Affected versions
- 4.0.2 - all
- 4.2.0 - all
- 4.3.0 - Electron is completely unstable >50% tests are broken. However, all tests are passing in Chrome version (at least on local)

## PS
Probably this is an Electron issue, but 4.0.2 was not working with Chrome as well.

**Comment by jennifer-shehane** [comment]
Unfortunately we'll have to close this issue as there is not enough information to reproduce the problem. This does not mean that your issue is not happening - it just means that we do not have a path to move forward. 

Can you provide any more information on how to reproduce this? Or some idea on what your tests are doing? The information provided is not specific enough to narrow down anything.

**Comment by theKashey** [comment]
>  it just means that we do not have a path to move forward.
We also have to way to move forward, and have no way to reproduce/understand the root cause. That's why I am exposing this problem _externally_.
However, it would be great if you will double check Cypress responsibilities and 

Further investigations:
[cypress-plugin-retries](https://github.com/Bkucera/cypress-plugin-retries) is capable to solve the problem, majority of our red test are not red during the second run.

However, when it is enabled:
- using Chrome as runner makes tests hung after a few minutes of work
- using Electron as a runner lead to out-of-memory exceptions after the same few minutes

More interesting:
- with 0 retries I could handle the page, and the subsequent page
- with 1 retries I could handle the page, and the subsequent page, and the subsequent page
- and visa versa. This is somehow related to the file cache.

Last time we had out-of-memory it was caused by frequent UI changes.

Running "red" tests on local is not leading to out-of-memory error, however, the heap is at 1600mb, which is probably over CI container limits.

**Comment by theKashey** [comment]
After tests ends:
- 300mb was stored in `snapshots` and related to CSS-in-JS. Setting `numTestsKeptInMemory` to `1` lowered that to 30mb.
- 34% are stored in "Arrays", and some big portion of it is stored in `metaMap`, especially referring some scripts, not letting them be free.

Ideally all data should be released after every "page" update, at least at CI, when you don't "need" it.

**Comment by jennifer-shehane** [comment]
I'm having a hard time following what the actual problem is that you are describing. 

The initial issue was ERR_INVALID_ARGUMENT - you have not provided a reproducible example for this, so we'll have to close the issue. 

The other comments are related to performance - particularly with the [cypress-plugin-retries](https://github.com/Bkucera/cypress-plugin-retries) - this is an outside plugin, is experimental, and outside the scope of the Cypress product, although we have our own retries functionality we are currently working on.

**Comment by theKashey** [comment]
At the end of the day we have solved our issue, more or less:
- `the root cause` - network failure for precached scripts. The reason is still unknown, unreproducible, we have only one project and just a few tests(of hundreds) which are affected (I tend to think this is an Electon bug)
- the mitigation - `cypress-plugin-retries` is able to "fix" broken tests
- the side effect - after a few retries cypress is failing with `out memory`  (this is purely Cypress bug, no extra memory is expected to be used on CI; numTestsKeptInMemory should have zero effect)
- the final solution - keeping test files small, as long as cypress runs each of them in isolation, and that is preventing heap from growing over limits (theoretically it could be added as advice to best practices - we had roughly "per feature" tests, and to "fix" our problem we separate them to at least 3 smaller pieces).

I hope no other user will face this issue, however if someone will - this issue might help at least a little bit. This is why it was created in the first place.

**Comment by stephsavoie** [comment]
@theKashey @jennifer-shehane 

I experienced the exact same issue as you since about a week.   Only difference:  I'm running Cypress 3.8.3.  On CI env (running Bamboo),  1-2 tests out of 135 were failing randomly and always on the very first command after `cy.visit()` (which is `cy.get()` in all failures analyzed), no elsewhere.  I'm running Vue 2.6.10.

It happened for the first time yesterday in interactive mode:  I saw all the requests being completed successfully, which means that page should be fully rendered for the user.  But Cypress test browser showed a blank page instead as response.

Still investigating, but I'm suspecting some blocked external api calls (like Google Analytics) which are not shown in the Cypress console but make it wait until timeout is reached (30 seconds).  I also suspect lazy-loaded routes to create random issues in Cypress.
